diesel_tests/tests/load_extension.rs

```diff
@@ -0,0 +1,30 @@
+use diesel::{sql_query, Connection, RunQueryDsl, SqliteConnection};
+
+fn conn() -> SqliteConnection {
+    SqliteConnection::establish(":memory:").unwrap()
+}
+
+#[test]
+fn test_load_extension() {
+    let conn = conn();
+
+    // disable extension loading
+    conn.enable_load_extension(false).unwrap();
+
+    // try loading module without enabling extension loading
+    let result = sql_query("SELECT load_extension('/tmp/foo');")
+        .execute(&conn)
+        .expect_err("should have been error");
+
+    assert_eq!(&format!("{}", &result), "not authorized");
+
+    // enable extension loading
+    conn.enable_load_extension(true).unwrap();
+
+    // try loading module without enabling extension loading
+    let result = sql_query("SELECT load_extension('/tmp/foo');")
+        .execute(&conn)
+        .expect_err("should have been error");
+
+    assert_ne!(&format!("{}", &result), "not authorized");
```

Comment:
Could we check here for the actual error message? I've looked like 2 minutes on it trying to figure out what is the difference to the check above. Even better would be to check the actual loading of a real extension.

Reply:
Yes, the actual message says something like 'cant load file' when extension loading is enabled.

The loading of an actual extension is problematic because that introduces a test dependency for all platforms, for which we don't really have any other use than this test. Then maintaining platform versions and hoping that the dependency won't break is a bit cumbersome.